frontend/src/app/core/helpers/exclusion-date-helper.ts

```diff
@@ -0,0 +1,150 @@
+import { convertTimeToOffset, getTimeString } from '@core/helpers/time-helper';
+import { TimeRangesMergeHelper } from '@core/helpers/time-ranges-merge-helper';
+import { getDateWithoutLocalOffset } from '@core/helpers/time-zone-helper';
+import { IDayTimeRange } from '@core/models/schedule/exclusion-date/IDayTimeRange';
+import { IExclusionDate } from '@core/models/schedule/exclusion-date/IExclusionDate';
+
+export const getExclusionDate = (
+    selected: string,
+    startHours: number,
+    startMinutes: string,
+    endHours: number,
+    endMinutes: string,
+): IExclusionDate => ({
+    selectedDate: selected,
+    dayTimeRanges: [
+        {
+            start: getTimeString(startHours, startMinutes),
+            end: getTimeString(endHours, endMinutes),
+        },
+    ],
+});
+
+export function getExclusionDateWithDayOffset(
+    exclusionDate: IExclusionDate,
+    dateAdapter: DateAdapter<Date>,
+    startHours: number,
+    startMinutes: string,
+    endHours: number,
+    endMinutes: string,
+    dayAction: DayAction,
+) {
+    let convertedDate: Date;
+
+    if (dayAction === DayAction.AddDay) {
+        convertedDate = dateAdapter.addCalendarDays(new Date(exclusionDate.selectedDate), 1);
+    } else if (dayAction === DayAction.SubtractDay) {
+        convertedDate = dateAdapter.addCalendarDays(new Date(exclusionDate.selectedDate), -1);
+    } else {
+        convertedDate = new Date(exclusionDate.selectedDate);
+    }
+
+    return [
+        getExclusionDate(
+            getDateWithoutLocalOffset(convertedDate).toJSON(),
+            startHours,
+            startMinutes,
+            endHours,
+            endMinutes,
+        ),
+    ];
+}
+
+export const getConvertedDatesForExclusionDate = (
+    exclusionDate: IExclusionDate,
+    dateAdapter: DateAdapter<Date>,
+    startHoursDayAction: DayAction,
+): [string, string] =>
+    (startHoursDayAction === DayAction.SubtractDay
+        ? [
+            getDateWithoutLocalOffset(dateAdapter.addCalendarDays(new Date(exclusionDate.selectedDate), -1)).toJSON(),
+            exclusionDate.selectedDate,
+        ]
+        : [
+            exclusionDate.selectedDate,
+            getDateWithoutLocalOffset(dateAdapter.addCalendarDays(new Date(exclusionDate.selectedDate), 1)).toJSON(),
+        ]);
+
+export const convertExclusionDateToOffset = (
+    exclusionDate: IExclusionDate,
+    timeZoneValue: string,
+    dateAdapter: DateAdapter<Date>,
+): IExclusionDate[] =>
+    exclusionDate.dayTimeRanges
+        .map((dayTimeRange) => {
+            const [startHours, startMinutes, startHoursDayAction] = convertTimeToOffset(
+                dayTimeRange.start,
+                timeZoneValue,
+            );
+            const [endHours, endMinutes, endHoursDayAction] = convertTimeToOffset(
+                dayTimeRange.end === '23:59' && !timeZoneValue.includes('00:00') ? '24:00' : dayTimeRange.end,
+                timeZoneValue,
```

Comment:
Looks very starnge

Reply:
The time range is stored as two strings and I manually convert them to a user time zone. Sometimes a time range can be divided into two parts: one for this day and part for the next day. I consider that day starts at 00:00 and ends at 23:59. It means that during conversion 22:00 can be converted to 23:59, but when I convert backwards, I need 24:00 value in order to make everything work correctly

Reply:
Fixed